crates/trigger-http/src/handler.rs

```diff
@@ -347,7 +347,8 @@ impl HandlerType {
 
 fn set_http_origin_from_request(
     store: &mut Store,
-    engine: &TriggerAppEngine<HttpTrigger>,
+    engine: &Arc<TriggerAppEngine<HttpTrigger>>,
+    handler: &HttpHandlerExecutor,
```

Comment:
We could instantiate it within the function I guess, but conceptually what this is saying is "I want to use the _exact same_ engine and executor that the HTTP trigger has set up."  That the executor has no state today is an implementation detail that this function shouldn't have to care about.

Really I should probably Arc it rather than cloning it, in case changes are made to the executor after I capture it...